pkg/apparmor/Dockerfile

```diff
@@ -1,28 +1,76 @@
 # syntax=docker/dockerfile-upstream:1.5.0-rc2-labs
 
-# Copyright (c) 2023 Zededa, Inc.
+# Copyright (c) 2023-2025 Zededa, Inc.
 # SPDX-License-Identifier: Apache-2.0
 
-FROM lfedge/eve-alpine:0f2e0da38e30753c68410727a6cc269e57ff74f2 as build
-ENV BUILD_PKGS linux-headers musl-dev musl-utils musl-libintl git gcc g++ \
-               autoconf automake libtool make flex bison bash sed gettext
+FROM --platform=${BUILDPLATFORM} lfedge/eve-alpine:0f2e0da38e30753c68410727a6cc269e57ff74f2 as builder-native-base
+ENV BUILD_PKGS="linux-headers musl-dev musl-utils musl-libintl git gcc g++ \
+               autoconf automake libtool make flex bison bash sed gettext"
 ENV PKGS alpine-baselayout
 RUN eve-alpine-deploy.sh
 
+FROM --platform=${BUILDPLATFORM} lfedge/eve-cross-compilers:fb809cfb1909752acb563e0b77cd3799534bce64 AS cross-compilers
+
+FROM builder-native-base as builder-cross-base
+COPY --from=cross-compilers /packages /packages
+
+FROM builder-cross-base as builder-target-arm64
+ARG COMPILER_TARGET_ARCH=aarch64
+
+FROM builder-cross-base as builder-target-amd64
+ARG COMPILER_TARGET_ARCH=x86_64
+
+FROM --platform=${TARGETPLATFORM} lfedge/eve-alpine:0f2e0da38e30753c68410727a6cc269e57ff74f2 as target-sysroot
+# Install the target sysroot
+ENV BUILD_PKGS="musl-dev libgcc musl-libintl libintl  linux-headers"
+RUN eve-alpine-deploy.sh
+
+#hadolint ignore=DL3006
+FROM builder-target-${TARGETARCH} as builder-target
+# install cross compiler
+#hadolint ignore=DL3006,DL3018
+RUN apk add --no-cache --allow-untrusted -X /packages "build-base-${COMPILER_TARGET_ARCH}"
+
+FROM builder-target as builder-amd64-arm64
+ENV CONFIGURE_TARGETS="--build=aarch64-alpine-linux-musl --host=x86_64-alpine-linux-musl"
+# copy libraries from target-sysroot
+COPY --from=target-sysroot /usr/lib/ /usr/x86_64-alpine-linux-musl/lib/
```

Comment:
instead of copying the whole /usr/lib and /usr/include you can use an approach like this to create the target libraries:
https://github.com/lf-edge/eve/blob/master/pkg/pillar/Dockerfile#L40

so you install only the libraries you need and they will be available at `/out`, you can even use a single code block to automate the process for the target architecture: https://github.com/lf-edge/eve/blob/master/pkg/pillar/Dockerfile#L53

Reply:
@rene sure, I actually did not mean to merge it to be honest, but I can fix it :+1: